zvariant/src/maybe.rs

```diff
@@ -103,6 +103,49 @@ impl<'a> Maybe<'a> {
     }
 }
 
+impl Display for Maybe<'_> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        maybe_display_fmt(self, f, true)
+    }
+}
+
+pub(crate) fn maybe_display_fmt(
+    maybe: &Maybe<'_>,
+    f: &mut std::fmt::Formatter<'_>,
+    type_annotate: bool,
+) -> std::fmt::Result {
+    if type_annotate {
+        write!(f, "@{} ", maybe.full_signature())?;
+    }
+
+    let (last_inner, depth) = {
+        let mut curr = maybe.inner();
+        let mut depth = 0;
+
+        while let Some(Value::Maybe(child_maybe)) = curr {
+            curr = child_maybe.inner();
+            depth += 1;
+        }
+
+        (curr, depth)
+    };
+
+    if let Some(last_inner) = last_inner {
+        // There are no Nothings, so print out the inner
+        // value with no prefixes.
```

Comment:
nit: can easily fit in the 100 chars limit. Same for other comments: break only when needed and at 100 chars limit.